### PR TITLE
Bugfix bh 5623

### PR DESCRIPTION
**Pull request by kaleongchan**
handle fail to find record

### CODE REVIEW

**Comment by kaleongchan** [comment]
> Looks good. I see you've decided to commit the lockfile. Might not be stricltly necessary since this is a library, not something that is deployed. but I guess it can't hurt either.

@darkbluesun undid the lock as it's not desirable.

**Comment by ngodinhloc** [comment]
it's good to see that my work has been picked up and continue

File: src/ORM/EntityManager.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ public function findBy(string $className = null, array $conditions = null, array
         $array = $this->mapper->toArray($entity);
         $builder = new Builder();
         $query = $builder->from($objectType)->select(array_keys($array))->where($conditions)->order($orders)->limit($limit)->getQuery();
-        $result = $this->connection->getClient()->query($query);
+        $result = $this->connection->getClient()->query($query) ?: [];
         $collections = $this->resultToCollection($result, $className, $lazy);
```
Comment (ngodinhloc) [inline src/ORM/EntityManager.php:108]: i think it would be better to handle the empty $result inside the function resultToCollection

Comment (kaleongchan) [reply src/ORM/EntityManager.php:108]: Hi Ken @ngodinhloc good to see you!
Handling it in resultToCollection means that we need to remove type hint, which we probably want to keep so it's more explicit.

**Comment by kaleongchan** [comment]
Please help merging this PR whoever has the permission. Thank you.

**Pull request merged**